docs/api/datamodules.rst

```diff
@@ -6,6 +6,9 @@ torchgeo.datamodules
 Geospatial DataModules
 ----------------------
 
+Sentinel2 NCCM
```

Comment:
Let's put this in a section called "Sentinel" and sort it alphabetically. Also needs `^^^` underline to make it a section title instead of text.